### ISSUE TITLE
Irregular exposure checks

### ISSUE THREAD

**Issue by tobi-coder** [DP-3T/dp3t-app-android-ch#162]
On my Huawei Mate 10 Pro (BLA-L29; Android 10) the apps seems to execute exposure checks very irregularly. It happens only 3 times (July 2nd, 4th and 16th). The check at the 16th July was forced by opening the Covid notification settings in the Android system settings .

Exported exposure checks:
`[{"timestamp":"Juli 16, 2020 19:17","keyCount":21,"matchesCount":0,"appName":"SwissCovid","hash":"ychfsT1+dMgTnum4cx6W\/NQucYn81woa6W98Nh4ia0U="},{"timestamp":"Juli 16, 2020 19:17","keyCount":32,"matchesCount":0,"appName":"SwissCovid","hash":"jftAi5ehWSKp6oXzL2yAZHnl7eh3vqB5w3ACElX9Mhw="},{"timestamp":"Juli 16, 2020 19:17","keyCount":44,"matchesCount":0,"appName":"SwissCovid","hash":"Y\/6oOQew2+qCER31kDZpj6nS2Z\/lyDB7YQQUZMZZl7I="},{"timestamp":"Juli 16, 2020 19:17","keyCount":54,"matchesCount":0,"appName":"SwissCovid","hash":"fEAOjHggxN0JlRIHQnV874pRJ6eM0Uw0jux81IVaVTo="},{"timestamp":"Juli 16, 2020 19:17","keyCount":57,"matchesCount":0,"appName":"SwissCovid","hash":"2KCkIYFlOZQJciXuUaFdyL2IDxhu2hlF0Ljv8xolNEc="},{"timestamp":"Juli 16, 2020 19:17","keyCount":52,"matchesCount":0,"appName":"SwissCovid","hash":"RJFKRgOEXEiVyU+Y0YyWVwhjmkoIXhizYbu9D2Afdl0="},{"timestamp":"Juli 16, 2020 19:17","keyCount":52,"matchesCount":0,"appName":"SwissCovid","hash":"FqbUp9KBH4afOR0UUvitD5HtcHKpTI9McABEFVcWV44="},{"timestamp":"Juli 16, 2020 19:17","keyCount":55,"matchesCount":0,"appName":"SwissCovid","hash":"7HO4IWCAoXeRHeCac18uysz+xmVfcE+iWcDMH2LSNWI="},{"timestamp":"Juli 16, 2020 19:17","keyCount":57,"matchesCount":0,"appName":"SwissCovid","hash":"2SP6WyuTEB3qCK1vcOtxki8vAH40Ltaia\/VESNNxHWM="},{"timestamp":"Juli 16, 2020 19:17","keyCount":54,"matchesCount":0,"appName":"SwissCovid","hash":"CoBai+kU1NjdeOiG1aEGoqag+iIbId7+TAGckLfWcDE="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"CW6eJliTeNU7pATrNofKs1iYo5EEjWmwxJKluuRqEAY="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"l6R9u1+24nt2cC4iiGI1Q4l0QBSYLIQ7VN7IzibaUKE="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"3VAfZNAAxVkkJW62tG0np1J+bMaPyXcYF5IDEnGaSYg="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"SYKLYA8cgloD5Nz71aTuA+KwEyaWNJ2MfXBoYo9+ChI="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"ePn6sAGwjY5pSpx7hDkkqhthW\/rfs8BZS2HL0bf9\/M0="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"3Eu+wwho5W9RhQnHxigEmOYq5Tzb91y3laWyR5L6AGU="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"dkmhbwUFeVD1l0Epmp5ZrwuCu3I9xARF5\/Mweyn77\/c="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"zCI+nM5xnJcupAWSqP8anYj3u+zyfVjAauRm+bFVbGQ="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"SJ33qPvm4jmapNE\/AxTJ4HlABYfuV6S5PEZ8qXyNNL8="},{"timestamp":"Juli 2, 2020 19:30","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"D0KDu3WjtUlVBbRNOSykAby2Xx18OYHt5rE6x4CwdyU="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"6MnEiHbBrrE18e7TjLGjJcDBjuQiQHhmXRvMSmkHJg4="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"vRSuzl4lhnw0B\/klElT+dX9OzHOrblJg6W+0bbCuK1k="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"89RxSKm57ts16a0GKz60j12+yOr0wGLYDZoq3eQvAn8="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"hvx30BXFCRKfpuT1yUT4kQHY7VnlrOc470nArSjM6Ho="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"exIksdkplR1sBfA7aFI+V19762HPOfK3HfEXL6fa+Nw="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"el5nsJW5hPqgDVaA63EznyUiM0xqAPCEHRxd5Pnt73c="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"prxBq1odcg0kA62p2prujgoXQdD8kfTlrDhxqeXh5gA="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"m0bjgpO4Xe+Si+BR+i4BdCFQil95nc4cs7IFnNzGy3c="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"Be3pfTtK7\/OV\/Bibt7ixKi8Rm6Cb6mEEpKzW7NFypJA="},{"timestamp":"Juli 4, 2020 19:07","keyCount":0,"matchesCount":0,"appName":"SwissCovid","hash":"Lvuc45ZhuauHPKIrG8DJnZzEAcHTSixWzyFRDiUmD\/g="}]`

**Comment by simonroesch** [comment]
Do you have any battery optimization / battery safer apps installed on the device?

**Comment by tobi-coder** [comment]
I have no special app for battery management installed. But I figured out that the default battery handling is set to automatically even if I choose "Akku-Optimierung ignorieren" during the installation (see attachments). I changed this settings now manually and will check the effects.

![Screenshot_20200718_174252_ch admin bag dp3t](https://user-images.githubusercontent.com/57704382/87856473-aead8d00-c91f-11ea-9de8-9ce787717403.jpg)

![Screenshot_20200718_174619_com huawei systemmanager](https://user-images.githubusercontent.com/57704382/87856474-aead8d00-c91f-11ea-8a14-5f3dd00a15bc.jpg)

**Comment by simonroesch** [comment]
Can you please check if this change  had any effect?

**Comment by tobi-coder** [comment]
It seems to work when the option for automatic battery management is disabled. It executes exposure checks once or twice a day.
But when reinstalling the app the battery options are still not set correctly after the installation.

**Comment by simonroesch** [comment]
With SwissCovid 1.0.9 we added an additional feature that should help defending the app against battery optimisation by the system. Could you be so kind and check the default behaviour on your device with the latest version of SwissCovid?

**Comment by tobi-coder** [comment]
I tried now version 1.0.9. The behavior is still the same. When installing the app the automatic battery management is not disabled.

**Comment by simonroesch** [comment]
Yes, the automatic battery management is out of our control and handled by the operating system. But regardless of your battery management settings as of version 1.0.9 you should see exposure checks at least once per day. Can you confirm this?
